### ISSUE TITLE
Alerts not Triggering

### ISSUE THREAD

**Issue by kzimmerm** [Graylog2/graylog2-server#6770]
We have a 4 node cluster setup and alerts are not triggering unless we have a sustained message  processing of >20 per second.

## Expected Behavior
When an event comes in an alert is triggered as soon as the message comes in.  In our case an email alert when a message contains "Started Network Time Service".

## Current Behavior
When the message comes in we may or may not receive an alert depending on how chatty our systems are being to our Graylog instance.  It can range from a minute later to whenever graylog gets busy enough to send it.  We've had the alert come in weeks later.  Our cluster is up and running but not production.   When setting up the Event Definition we can see the message in the Filter Preview but no alert is ever generated until our graylog server gets busy.  When we enable the Random HTTP Message Generator then the alerts will come through in less than a minute.

## Possible Solution
Fix what ever process logic was added in to version 3.1 to alerts.  We can't have an alert missed in the middle of the night when our systems are less chatty.  That just doesn't make sense for an alert system.  Continuously spamming our Graylog server is not ideal or helpful.

## Steps to Reproduce (for bugs)
1. Setup 4 node cluster
2. Forward a servers's Syslog to Graylog
3. Create an Event Definition and Alert
4. Trigger the Event and watch the Alert is never created
5. Enable the Random HTTP Message Generator and the alert will come through shortly after.

## Context
We need proper alerting for events that come in when they come in.

## Your Environment
4 Node Centos 7.5 Cluster

* Graylog Version: 3.1.3
* Elasticsearch Version: 6.8.3-1
* MongoDB Version: 4.0.12-1
* Operating System:  CentOS Linux release 7.5.1804
* Browser version: Any
[server.txt](https://github.com/Graylog2/graylog2-server/files/3838157/server.txt)

I've attached our server config (from the master)

There are posts related to this [here](https://community.graylog.org/t/alerts-events-do-not-trigger/11792/48) and [here](https://github.com/Graylog2/graylog2-server/issues/6453)

**Comment by romanov-dv** [comment]
same problem
_processing_status_journal_write_rate_threshold = 0_ did not help
Graylog Version: 3.1.3

**Comment by kzimmerm** [comment]
When we enable the Random HTTP Generator on all nodes (Globally) then previous alerts that were not triggered end up coming through even if the alert was actually triggered days or weeks ago.

Enabling the Random HTTP Generator (locally) on our main node that receives the majority of the messages as well as being our graylog master does nothing to help.  It's only when the input is marked as Global do alerts come through.

Initially we don't see the alerts in the GL_EVENTS index.   After we enable the HTTP generator (globally) then the alerts show up.  We tried setting processing_status_journal_write_rate_threshold to 0 on master only, all nodes and it didn't help.

**Comment by kzimmerm** [comment]
We see this every second in the log when doing a grep for AggregationEventProcessor


`2019-11-26T10:53:24.938-06:00 DEBUG [AggregationEventProcessor] Creating events for config=AggregationEventProcessorConfig{type=aggregation-v1, query="Started Network Time Service", streams=[], groupBy=[], series=[], conditions=Optional[AggregationConditions{expression=Optional.empty}], searchWithinMs=60000, executeEveryMs=60000} parameters=AggregationEventProcessorParameters{type=aggregation-v1, timerange=AbsoluteRange{type=absolute, from=2019-11-23T20:42:07.123Z, to=2019-11-23T20:43:07.122Z}, streams=[], batchSize=500}
2019-11-26T10:53:25.067-06:00 DEBUG [JobExecutionEngine] Execute job: Started Network Time Service/5dc48ed291a3a71888a80e00/event-processor-execution-v1 (job-class=EventProcessorExecutionJob trigger=5dc48ed291a3a71888a80e01 config=Config{type=event-processor-execution-v1, eventDefinitionId=5dc48ed291a3a71888a80dff, parameters=AggregationEventProcessorParameters{type=aggregation-v1, timerange=AbsoluteRange{type=absolute, from=2019-11-26T16:22:26.466Z, to=2019-11-26T16:23:26.466Z}, streams=[], batchSize=500}, processingWindowSize=60000, processingHopSize=60000})
2019-11-26T10:53:25.366-06:00 DEBUG [AggregationEventProcessor] Creating events for config=AggregationEventProcessorConfig{type=aggregation-v1, query="Started Network Time Service", streams=[], groupBy=[], series=[], conditions=Optional[AggregationConditions{expression=Optional.empty}], searchWithinMs=60000, executeEveryMs=60000} parameters=AggregationEventProcessorParameters{type=aggregation-v1, timerange=AbsoluteRange{type=absolute, from=2019-11-23T20:43:07.123Z, to=2019-11-23T20:44:07.122Z}, streams=[], batchSize=500}
2019-11-26T10:53:25.519-06:00 DEBUG [JobExecutionEngine] Execute job: Started Network Time Service/5dc48ed291a3a71888a80e00/event-processor-execution-v1 (job-class=EventProcessorExecutionJob trigger=5dc48ed291a3a71888a80e01 config=Config{type=event-processor-execution-v1, eventDefinitionId=5dc48ed291a3a71888a80dff, parameters=AggregationEventProcessorParameters{type=aggregation-v1, timerange=AbsoluteRange{type=absolute, from=2019-11-26T16:22:26.466Z, to=2019-11-26T16:23:26.466Z}, streams=[], batchSize=500}, processingWindowSize=60000, processingHopSize=60000})
2019-11-26T10:53:25.817-06:00 DEBUG [AggregationEventProcessor] Creating events for config=AggregationEventProcessorConfig{type=aggregation-v1, query="Started Network Time Service", streams=[], groupBy=[], series=[], conditions=Optional[AggregationConditions{expression=Optional.empty}], searchWithinMs=60000, executeEveryMs=60000} parameters=AggregationEventProcessorParameters{type=aggregation-v1, timerange=AbsoluteRange{type=absolute, from=2019-11-23T20:44:07.123Z, to=2019-11-23T20:45:07.122Z}, streams=[], batchSize=500}
2019-11-26T10:53:25.945-06:00 DEBUG [JobExecutionEngine] Execute job: Started Network Time Service/5dc48ed291a3a71888a80e00/event-processor-execution-v1 (job-class=EventProcessorExecutionJob trigger=5dc48ed291a3a71888a80e01 config=Config{type=event-processor-execution-v1, eventDefinitionId=5dc48ed291a3a71888a80dff, parameters=AggregationEventProcessorParameters{type=aggregation-v1, timerange=AbsoluteRange{type=absolute, from=2019-11-26T16:22:26.466Z, to=2019-11-26T16:23:26.466Z}, streams=[], batchSize=500}, processingWindowSize=60000, processingHopSize=60000})
2019-11-26T10:53:26.243-06:00 DEBUG [AggregationEventProcessor] Creating events for config=AggregationEventProcessorConfig{type=aggregation-v1, query="Started Network Time Service", streams=[], groupBy=[], series=[], conditions=Optional[AggregationConditions{expression=Optional.empty}], searchWithinMs=60000, executeEveryMs=60000} parameters=AggregationEventProcessorParameters{type=aggregation-v1, timerange=AbsoluteRange{type=absolute, from=2019-11-23T20:45:07.123Z, to=2019-11-23T20:46:07.122Z}, streams=[], batchSize=500}
2019-11-26T10:53:26.373-06:00 DEBUG [JobExecutionEngine] Execute job: Started Network Time Service/5dc48ed291a3a71888a80e00/event-processor-execution-v1 (job-class=EventProcessorExecutionJob trigger=5dc48ed291a3a71888a80e01 config=Config{type=event-processor-execution-v1, eventDefinitionId=5dc48ed291a3a71888a80dff, parameters=AggregationEventProcessorParameters{type=aggregation-v1, timerange=AbsoluteRange{type=absolute, from=2019-11-26T16:22:26.466Z, to=2019-11-26T16:23:26.466Z}, streams=[], batchSize=500}, processingWindowSize=60000, processingHopSize=60000})
`

**Comment by jllorente25** [comment]
Your observation has helped me. I hadn't known how to make the alerts go off for months. I also have a four node system. All 4 are part of the ES cluster. I had some global syslog inputs, but really in the origins it only had it configured to be sent to 2 machines. 

We have tried to create that random global http source as you mentioned, and the alerts work!

As an alternative, we have tried to configure in origin, that the 4 machines are recipients of logs, and thus, with random input stopped, the alerts also work. 

Something that did not work was configuring the global inputs we had, to local for two nodes that were log receivers.

I don't think it had to do with the amount of logs to receive, we were receiving >1000 logs/sec

So the operating logic (i don't understand) is that the 4 nodes of the cluster have to be log receivers.

We have now graylog 4.7, elastic 7, mongo 4.4, but before, graylog 4.0, mongo 4.0.. and it didn't work either, it's not a version thing.

So thanks for your idea, but I don't really understand why this happens
